webrender/build_gfx.rs

```diff
@@ -28,18 +27,12 @@ const VK_EXTENSIONS: &'static str = "#extension GL_ARB_shading_language_420pack
 // https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#features-limits
 const MAX_INPUT_ATTRIBUTES: u32 = 16;
 
-const DESCRIPTOR_SET_PER_FRAME: usize = 0;
-const DESCRIPTOR_SET_SAMPLER: usize = 1;
-const DESCRIPTOR_SET_PER_DRAW: usize = 2;
-#[cfg(not(feature = "push_constants"))]
-const DESCRIPTOR_SET_LOCALS: usize = 3;
-#[cfg(feature = "push_constants")]
-const DESCRIPTOR_SET_COUNT: usize = 3;
-#[cfg(not(feature = "push_constants"))]
+const DESCRIPTOR_SET_PER_PASS: usize = 0;
+const DESCRIPTOR_SET_PER_FRAME: usize = 1;
```

Comment:
I assigned the `0` index to this set, because it's only used in a subset of shaders, so I can skip it when I bind descriptor sets to the pipeline layout by setting the first descriptor set index to `1` if the pipeline layout doesn't contains this set: https://github.com/szeged/webrender/pull/291/files#diff-fa2ec06e095c83d43cdc6e509e90c761R547

Reply:
when you bind descriptor sets *within* a pass, you wouldn't need to touch any of these two anyway